tools/protocol-rpc.sh

```diff
@@ -0,0 +1,234 @@
+ENV_FILE=".env"
+# ANVIL_PID_FILE=".anvil.pid"
+DEPLOYER_PRIVATE_KEY="0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"
+TEST_PRIVATE_KEY="TEST_PRIVATE_KEY=$DEPLOYER_PRIVATE_KEY"
+RPC_URL="http://127.0.0.1:8545"
+WAIT=false # Default to false
+ORCHESTRATOR_FACTORY_ADDRESS=""
+ERC20_MOCK_ADDRESS=""
+BANCOR_FORMULA_ADDRESS=""
+
+# Function to detect OS
+detect_os() {
+    local os
+    os=$(uname)
+    if [[ "$os" == "Darwin" ]]; then
+        printf "Detected macOS 🍏\n"
+    elif [[ "$os" == "Linux" ]]; then
+        printf "Detected Linux 🐧\n"
+    else
+        printf "Unsupported OS detected ❌\n" >&2
+        return 1
+    fi
+}
+
+# Function to check if dependencies are installed
+check_dependencies() {
+    printf "Checking for required dependencies 🔍\n"
+    for dep in "${DEPS[@]}"; do
+        if ! command -v "$dep" &>/dev/null; then
+            printf "$dep is not installed ❌\n"
+            install_dependency "$dep"
+        else
+            printf "$dep is already installed ✅\n"
+        fi
+    done
+}
+
+# Function to install dependencies
+install_dependency() {
+    local dep="$1"
+    printf "Attempting to install $dep ⚙️\n"
+    case "$dep" in
+    curl)
+        if [[ "$(uname)" == "Darwin" ]]; then
+            brew install curl || return 1
+        else
+            sudo apt-get install -y curl || return 1
+        fi
+        ;;
+    make)
+        if [[ "$(uname)" == "Darwin" ]]; then
+            xcode-select --install || return 1
+        else
+            sudo apt-get install -y build-essential || return 1
+        fi
+        ;;
+    forge | anvil)
+        printf "Installing Foundry 🛠️\n"
+        if ! curl -L "$INSTALL_URL" | bash; then
+            printf "Failed to install Foundry ❌\n" >&2
+            return 1
+        fi
+        source "$HOME/.bashrc" || source "$HOME/.zshrc"
+        foundryup
+        ;;
+    *)
+        printf "Unknown dependency $dep ❓\n" >&2
+        return 1
+        ;;
+    esac
+    printf "$dep installation completed 🚀\n"
+}
+
+# Function to run anvil
+start_anvil() {
+    printf "Starting Anvil blockchain simulator 🔥\n"
+    pkill anvil
+    anvil &
+    ANVIL_PID=$!
+    # echo "$ANVIL_PID" >"$ANVIL_PID_FILE"
+}
+
+# Function to handle contract installation
+install_contracts() {
+    printf "Running 'make install' ⚙️\n"
+    if ! make install; then
+        printf "Contract installation failed ❌\n" >&2
+        return 1
+    fi
+    printf "Inverter Protocol Installation Complete 🚀\n"
+}
+
+# Function to deploy the protocol
+deploy_protocol() {
+    printf "Starting Protocol Deployment 🚀\n"
+    source dev.env || {
+        printf "Failed to source dev.env ❌\n" >&2
+        return 1
+    }
+
+    # Capture deployment output and log to console
+    local deploy_output
+    if ! deploy_output=$(DEPLOYER_PRIVATE_KEY="$DEPLOYER_PRIVATE_KEY" forge script "$DEPLOY_SCRIPT" --rpc-url "$RPC_URL" --broadcast -v 2>&1 | tee /dev/tty); then
```

Comment:
Why do we require the user to set a pk? Can we not just use anvil's default account?

Reply:
@fabianschu, hello sir - did not leave any instructions because the flow does not need much more than `git submodule update --remote` and then `bun test` the script `bun run protocol-rpc` is for deploying the protocol for other reasons such as testing with the Control Room

Reply:
We don't require the user to set a PK its handled by the script.
- The scripts sets the PK in .env